### PR TITLE
Feature/site theme

### PR DESCRIPTION
**Pull request by jameswburke**
## Issue(s): Relates to or closes...
[IRV-486](https://alleyinteractive.atlassian.net/browse/IRV-486)

## Summary: This pull request will...
 This commit implements a new SiteThemeProvider which acts as a middleware component between the siteTheme provider setup in the backend, and allows those theme values to be set in css values set by the `style` prop.

## Tests: I know this code works because...
Tested locally.

### CODE REVIEW

File: packages/styled/components/hoc/withThemes.js [hunk 1]
```diff
@@ -1,18 +1,34 @@
...
+    // `color: #000000`.
+    const themeContext = useContext(ThemeContext);
+    const newProps = props;
+    Object.keys(style).forEach((property) => {
+      newProps.style[property] = get(
+        themeContext,
+        newProps.style[property],
+        newProps.style[property]
+      );
+    });
```
Comment (pattiereaves) [inline packages/styled/components/hoc/withThemes.js:28]: A reducer makes it more clear what the themeContext will replace in the style prop.

```suggestion
const newStyle = Object.keys(style)
  .reduce((acc, property) => {
    const themeStyles = get(
      themeContext,
      style[property],
      style[property] // is this right, should the key be the default value?
    );
    
    return {
      ...acc,
      [property]: themeStyles,
    };
  }, style);
const newProps = { ...props, style: newStyle };
```

File: packages/styled/components/hoc/withThemes.js [hunk 2]
```diff
@@ -21,10 +37,18 @@ const withThemes = (themeMap) => (WrappedComponent) => {
      * Prop indicating which theme to use.
      */
     themeName: PropTypes.oneOf(Object.keys(themeMap)),
+    /**
+     * CSS style.
+     */
+    style: PropTypes.oneOf([
+      PropTypes.array,
```
Comment (pattiereaves) [inline packages/styled/components/hoc/withThemes.js:44]: Will this still work if style is an array?

Comment (jameswburke) [reply packages/styled/components/hoc/withThemes.js:44]: It's an issue on the PHP side,

An empty (`[]`) or indexed (`[0=>'', 1=>'']`) array will be an array in JSON, and associate arrays (`['hello'=>'world', 'foo'=>'bar']`) will be an object in JSON. So if `style` is empty or unset on the WP side, it comes over as an array, which throws a propType error.

We're fixing on the WP side, but it's part of a much bigger thing and isn't ready yet. So you're right to call it out, but it's a temporary thing.

File: packages/styled/components/hoc/withThemes.js (hunk 1 above)
Comment (jameswburke) [reply packages/styled/components/hoc/withThemes.js]: Thanks, this makes sense.

**Pull request merged**